File: src/utils/logger.py
import logging
from .env import env

logger = logging.getLogger(__name__)
logger.setLevel(logging.DEBUG if env.get("DEBUG") else logging.INFO)

handler = logging.StreamHandler()
handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))

logger.addHandler(handler)

File: src/utils/logger.ts
const logger = {
  debug: (message: string, ...meta: any[]) => {
    console.log(JSON.stringify({ level: "debug", message, ...meta }));
  },
  info: (message: string, ...meta: any[]) => {
    console.log(JSON.stringify({ level: "info", message, ...meta }));
  },
  warn: (message: string, ...meta: any[]) => {
    console.warn(JSON.stringify({ level: "warn", message, ...meta }));
  },
  error: (message: string | Error, ...meta: any[]) => {
    if (message instanceof Error) {
      console.error(
        JSON.stringify({
          level: "error",
          message: message.message,
          stack: message.stack,
          ...meta,
        }),
      );
    } else {
      console.error(JSON.stringify({ level: "error", message, ...meta }));
    }
  },
};

export default logger;
